cmd/crossplane/stack/unpack/unpack.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unpack
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane/pkg/stacks"
+	"github.com/crossplane/crossplane/pkg/stacks/walker"
+)
+
+// Command configuration for unpacking stacks.
+type Command struct {
+	Name                      string
+	Dir                       string
+	OutFile                   string
+	PermissionScope           string
+	TemplatingControllerImage string
+}
+
+// FromKingpin produces a stack unpack command from a Kingpin command.
+func FromKingpin(cmd *kingpin.CmdClause) Command {
+	c := Command{Name: cmd.FullCommand()}
+	cmd.Flag("content-dir", "The absolute path of the directory that contains the stack contents").Required().StringVar(&c.Dir)
+	cmd.Flag("outfile", "The file where the YAML Stack record and CRD artifacts will be written").StringVar(&c.OutFile)
+	cmd.Flag("permission-scope", "The permission-scope that the stack must request (Namespaced, Cluster)").Default("Namespaced").StringVar(&c.PermissionScope)
```

Comment:
The change from String to StringVar may have introduced a new problem:
```
$ kubectl logs -n crossplane-system stack-gcp-lz4zd
crossplane: error: cannot unpack stack: failed to unpack stacks: Stack permissionScope "Cluster" is not permitted by unpack invocation parameters (expected "")
$ kubectl logs -n precipice stack-wordpress-fdxxq 
crossplane: error: cannot unpack stack: failed to unpack stacks: Stack permissionScope "Namespaced" is not permitted by unpack invocation parameters (expected "")
```

Reply:
Good catch - the problem was actually that I should have been returning struct pointers. The flags weren't actually being set in the `Command` structs, so they were all using default values.